### PR TITLE
Update for v0.16.0.

### PR DESCRIPTION
**Pull request by pp-mo**
For latest release https://github.com/SciTools/iris-grib/releases/tag/v0.16.0

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by pp-mo** [comment]
@conda-forge-admin, please rerender

**Comment by pp-mo** [comment]
@conda-forge-admin, please rerender

**Comment by lbdreyer** [comment]
Thanks @pp-mo  !

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ test:
 
 about:
   home: https://iris-grib.readthedocs.io
-  license: LGPL-3.0
+  license: LGPL-3.0-only
```
Comment (lbdreyer) [inline recipe/meta.yaml]: @pp-mo 
Why `only`?
I notice that Iris has `LGPL-3.0-or-later`, should they not be the same?

Comment (bjlittle) [reply recipe/meta.yaml]: Yup, go with `LGPL-3.0-or-later` :+1:

File: recipe/meta.yaml [hunk 2]
```diff
@@ -19,7 +19,7 @@ requirements:
     - pip
   run:
     - python
-    - iris >=2
+    - iris >=3
     - python-eccodes
```
Comment (bjlittle) [inline recipe/meta.yaml:23]: @lbdreyer suggests a minimum pin here for `python-eccodes`

File: recipe/meta.yaml (hunk 1 above)
Comment (bjlittle) [inline recipe/meta.yaml]:
```suggestion
  license: LGPL-3.0-or-later
```

File: recipe/meta.yaml (hunk 2 above)
Comment (bjlittle) [inline recipe/meta.yaml:23]:
```suggestion
    - python-eccodes
    - eccodes >=2.16
```

**Pull request merged**